lib/modules/datasource/maven/util.ts

```diff
@@ -91,13 +93,78 @@ export async function downloadHttpProtocol(
       // istanbul ignore next
       logger.debug({ failedUrl }, 'Unsupported host');
     } else {
-      logger.info({ failedUrl, err }, 'Unknown error');
+      logger.info({ failedUrl, err }, 'Unknown HTTP download error');
     }
     return {};
   }
 }
 
-export async function checkHttpResource(
+function isS3CedentialsError(err: { name: string; message: string }): boolean {
+  return err.name === 'CredentialsProviderError';
+}
+
+function isS3RegionError(err: { name: string; message: string }): boolean {
+  return err.message === 'Region is missing';
```

Comment:
do we really need those simple wrapper functions? i think that can be inlined or the whole error checking should be done in one helper function.

Reply:
I followed the existing HTTP implementation for consistency.

Reply:
ok, don't do that for such simple one-line. This simply blows up code and makes reviews more complicated